src/classes/index.js

```
@@ -48,12 +45,16 @@ export function getAllClasses(client) {
     return client.query(q.Map(q.Paginate(Ref("classes")), clazz => q.Get(clazz)))
       .then(result => dispatch(updateClassInfo(result.data)))
       .then(() => dispatch(fetchingClasses(false)))
+      .catch(error => {
+        dispatch(fetchingClasses(false))
+        throw error
```

According with the [documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/catch):

> The Promise returned by catch() is rejected if `onRejected` throws an error or returns a Promise which is itself rejected; otherwise, it is resolved.

So, we can either throw the exception again or return a new promise rejected. Chaining errors is a bit different than a normal chain.

https://www.peterbe.com/plog/chainable-catches-in-a-promise